receiver/sqlqueryreceiver/config.go

```diff
@@ -28,16 +30,121 @@ type Config struct {
 	Queries                                 []Query `mapstructure:"queries"`
 }
 
+func (c Config) Validate() error {
+	if c.Driver == "" {
+		return errors.New("'driver' cannot be empty")
+	}
+	if c.DataSource == "" {
+		return errors.New("'datasource' cannot be empty")
+	}
+	if len(c.Queries) == 0 {
+		return errors.New("'queries' cannot be empty")
+	}
+	for _, query := range c.Queries {
+		if err := query.Validate(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
 type Query struct {
-	SQL     string   `mapstructure:"sql"`
-	Metrics []Metric `mapstructure:"metrics"`
+	SQL     string      `mapstructure:"sql"`
+	Metrics []MetricCfg `mapstructure:"metrics"`
+}
+
+func (q Query) Validate() error {
+	if q.SQL == "" {
+		return errors.New("'query.sql' cannot be empty")
+	}
+	if len(q.Metrics) == 0 {
+		return errors.New("'query.metrics' cannot be empty")
+	}
+	for _, metric := range q.Metrics {
+		if err := metric.Validate(); err != nil {
```

Comment:
Would aggregating in a multierr be helpful to report more issues at once?

Reply:
Good idea -- will fix.

Reply:
Done.